tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -40,23 +43,31 @@ private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
 		gcdw.finishXML();
 	}
 
+
 	private void describeFeatureType(PrintWriter out, HttpServletRequest hsreq) {
 		WFSDescribeFeatureTypeWriter dftw = new WFSDescribeFeatureTypeWriter(out);
 		dftw.startXML();
 		dftw.setServer(hsreq.getScheme() + "://" + hsreq.getServerName() + ":" + hsreq.getServerPort() + "/thredds/wfs");
-		ArrayList<WFSFeatureAttribute> attributes = new ArrayList<>();
-		attributes.add(new WFSFeatureAttribute("catchments_geometry_container", "int"));
-		attributes.add(new WFSFeatureAttribute("hruid", "int"));
-		attributes.add(new WFSFeatureAttribute("lat", "double"));
-		attributes.add(new WFSFeatureAttribute("lon", "double"));
-		attributes.add(new WFSFeatureAttribute("catchments_area", "double"));
-		attributes.add(new WFSFeatureAttribute("catchments_perimeter", "double"));
-		attributes.add(new WFSFeatureAttribute("catchments_veght", "double"));
-		attributes.add(new WFSFeatureAttribute("catchments_cov", "double"));
-		dftw.addFeature(new WFSFeature("hru_soil_moist", "HRU Soil Moisture", "simplegeom",attributes));
-		dftw.writeFeatures();
-		dftw.finishXML();
+		try {
+
+			NetcdfDataset data = NetcdfDataset.openDataset("../cdm/src/test/data/dataset/SimpleGeos/hru_soil_moist_3hru_5timestep.nc");

```

Comment:
Good start! To remove hardcoding, you could get the dataset that the user is interested in from the HttpServletRequest. Potentially relevant Javadocs include:
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getParameter-java.lang.String-
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getParameterMap--
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getParameterNames--
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getParameterValues-java.lang.String-

The names of the parameters you should expect on the request are defined by the WFS Specification for the DescribeFeatureType operation.

Informal intro:
http://cite.opengeospatial.org/pub/cite/files/edu/wfs/text/operations.html#describefeaturetype

Another informal intro from GeoServer:
http://docs.geoserver.org/latest/en/user/services/wfs/reference.html#describefeaturetype

Formal spec (too long, don't read in its entirety):
http://portal.opengeospatial.org/files/?artifact_id=39967

One thing to look out for is that the parameter names are usually case-insensitive. There are a few examples of case-insensitive handling of parameters retrieved from `HttpServletRequest`s in this class (`WFSController`) already. Those might serve as inspiration.